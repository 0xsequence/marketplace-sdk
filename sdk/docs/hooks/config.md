# config

## References

### useConfig

Re-exports [useConfig](config/useConfig.md#useconfig)

***

### useConnectorMetadata

Re-exports [useConnectorMetadata](config/useConnectorMetadata.md#useconnectormetadata)

***

### useMarketplaceConfig

Re-exports [useMarketplaceConfig](config/useMarketplaceConfig.md#usemarketplaceconfig)
