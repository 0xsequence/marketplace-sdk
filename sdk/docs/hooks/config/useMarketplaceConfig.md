# config/useMarketplaceConfig

## Functions

### useMarketplaceConfig()

```ts
function useMarketplaceConfig(): UseQueryResult<MarketplaceConfig, Error>;
```

Defined in: [sdk/src/react/hooks/config/useMarketplaceConfig.tsx:5](https://github.com/0xsequence/marketplace-sdk/blob/6a4808051b4d56769c8daea217398414041a4d84/sdk/src/react/hooks/config/useMarketplaceConfig.tsx#L5)

#### Returns

`UseQueryResult`\<`MarketplaceConfig`, `Error`\>
