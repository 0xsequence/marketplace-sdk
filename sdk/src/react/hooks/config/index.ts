export * from './useConfig';
export * from './useConnectorMetadata';
export * from './useCurrency';
export * from './useMarketCurrencies';
export * from './useMarketplaceConfig';
