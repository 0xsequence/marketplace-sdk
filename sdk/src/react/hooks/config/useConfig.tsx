'use client';

import { useContext } from 'react';

import { MarketplaceProviderNotFoundError } from '../../../utils/errors';
import { MarketplaceSdkContext } from '../../providers';

export function useConfig() {
	const context = useContext(MarketplaceSdkContext);
	if (!context) {
		throw new MarketplaceProviderNotFoundError();
	}
	return context;
}
