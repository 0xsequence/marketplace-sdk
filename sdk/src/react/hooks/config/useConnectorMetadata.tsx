import { useAccount } from 'wagmi';
import { WalletKind } from '../../_internal/api';

/**
 * Provides metadata about the currently connected wallet connector
 *
 * This hook analyzes the active wallet connection to determine the wallet type
 * and specific characteristics. It's particularly useful for implementing
 * wallet-specific features or UI adjustments based on whether the user is
 * connected via Sequence wallet, WaaS (Wallet as a Service), or other wallets.
 *
 * @returns Wallet connector metadata
 * @returns returns.isWaaS - True if connected via Wallet as a Service
 * @returns returns.isSequence - True if connected via any Sequence wallet variant
 * @returns returns.walletKind - Enum indicating the specific wallet type (sequence or unknown)
 *
 * @example
 * Basic usage:
 * ```typescript
 * const { isWaaS, isSequence, walletKind } = useConnectorMetadata();
 *
 * if (isWaaS) {
 *   console.log('User is connected via WaaS');
 * }
 * ```
 *
 * @example
 * Conditional rendering based on wallet type:
 * ```typescript
 * const { isSequence, walletKind } = useConnectorMetadata();
 *
 * return (
 *   <div>
 *     {isSequence ? (
 *       <SequenceWalletFeatures />
 *     ) : (
 *       <GenericWalletInterface />
 *     )}
 *     <p>Wallet type: {walletKind}</p>
 *   </div>
 * );
 * ```
 *
 * @remarks
 * This hook depends on wagmi's useAccount hook and will return default values
 * when no wallet is connected. The detection is based on the connector ID string
 * patterns, where 'waas' suffix indicates WaaS and 'sequence' substring indicates
 * any Sequence wallet variant.
 *
 * @see {@link useAccount} - The underlying wagmi hook for account information
 * @see {@link WalletKind} - Enum defining possible wallet types
 */
export const useConnectorMetadata = () => {
	const { connector } = useAccount();

	const isWaaS = connector?.id.endsWith('waas') ?? false;
	const isSequence = connector?.id.includes('sequence');
	const walletKind = isSequence ? WalletKind.sequence : WalletKind.unknown;

	return {
		isWaaS,
		isSequence,
		walletKind,
	};
};
