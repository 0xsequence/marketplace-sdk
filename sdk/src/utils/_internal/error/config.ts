import { BaseError } from './base';

export type ConfigErrorType<name extends string = 'ConfigError'> = BaseError & {
	name: name;
};

export class ConfigError extends BaseError {
	override name = 'ConfigError';
}

export class InvalidProjectAccessKeyError extends ConfigError {
	override name = 'InvalidProjectAccessKeyError';
	constructor(projectAccessKey: string) {
		super(`Invalid project access key: ${projectAccessKey}`);
	}
}
