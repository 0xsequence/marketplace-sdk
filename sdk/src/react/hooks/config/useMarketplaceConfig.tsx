import { useQuery } from '@tanstack/react-query';
import { marketplaceConfigOptions } from '../../queries/marketplaceConfig';
import { useConfig } from '../config/useConfig';

export const useMarketplaceConfig = () => {
	const config = useConfig();
	return useQuery(marketplaceConfigOptions(config));
};
