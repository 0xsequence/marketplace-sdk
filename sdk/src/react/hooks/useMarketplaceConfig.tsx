import { useQuery } from '@tanstack/react-query';
import { marketplaceConfigOptions } from '../queries/marketplaceConfig';
import { useConfig } from './useConfig';

export const useMarketplaceConfig = () => {
	const config = useConfig();
	return useQuery(marketplaceConfigOptions(config));
};
