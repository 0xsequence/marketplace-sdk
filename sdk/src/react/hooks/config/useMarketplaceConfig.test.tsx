import { renderHook, server, waitFor } from '@test';
import { describe, expect, it } from 'vitest';
import { createLookupMarketplaceErrorHandler } from '../../../../_internal/api/__mocks__/builder.msw';

import { useMarketplaceConfig } from '../../config/useMarketplaceConfig';

describe('useMarketplaceConfig', () => {
	it('should fetch marketplace config and styles successfully', async () => {
		const { result } = renderHook(() => useMarketplaceConfig());

		// Initially loading
		expect(result.current.isLoading).toBe(true);
		expect(result.current.data).toBeUndefined();

		// Wait for data to be loaded
		await waitFor(() => {
			expect(result.current.isLoading).toBe(false);
		});

		// Verify the data matches our mock
		expect(result.current.data).toBeDefined();
		expect(result.current.data).toMatchSnapshot();
		expect(result.current.error).toBeNull();
	});

	it('should handle config fetch error', async () => {
		// Override the handler for this test to return an error
		server.use(createLookupMarketplaceErrorHandler());

		const { result } = renderHook(() => useMarketplaceConfig());

		await waitFor(() => {
			expect(result.current.isError).toBe(true);
		});

		expect(result.current.error).toBeDefined();
		expect(result.current.data).toBeUndefined();
	});
});
