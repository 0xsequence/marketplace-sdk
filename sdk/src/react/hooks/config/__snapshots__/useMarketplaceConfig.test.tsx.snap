// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useMarketplaceConfig > should fetch marketplace config and styles successfully 1`] = `
{
  "market": {
    "bannerUrl": "https://example.com/market-banner.png",
    "collections": [
      {
        "bannerUrl": "https://example.com/market-banner.png",
        "chainId": 1,
        "contractType": "ERC721",
        "createdAt": "2025-03-16T13:04:16.098Z",
        "currencyOptions": [
          "0x0000000000000000000000000000000000000000",
          "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
        ],
        "destinationMarketplace": "sequence_marketplace_v2",
        "feePercentage": 3.5,
        "filterSettings": {
          "exclusions": [
            {
              "condition": "SPECIFIC_VALUE",
              "key": "Type",
              "value": "Sample",
            },
          ],
          "filterOrder": [
            "Type",
            "Rarity",
          ],
        },
        "id": 1,
        "itemsAddress": "0x0000000000000000000000000000000000000000",
        "marketplaceType": "market",
        "private": false,
        "projectId": 1,
        "updatedAt": "2025-03-16T13:04:16.098Z",
      },
      {
        "bannerUrl": "https://example.com/collection-banner.png",
        "chainId": 137,
        "contractType": "ERC1155",
        "createdAt": "2025-03-16T13:04:16.098Z",
        "currencyOptions": [
          "0x0000000000000000000000000000000000000000",
        ],
        "destinationMarketplace": "opensea",
        "feePercentage": 2.5,
        "filterSettings": {
          "exclusions": [
            {
              "condition": "ENTIRE_KEY",
              "key": "Category",
            },
            {
              "condition": "SPECIFIC_VALUE",
              "key": "Level",
              "value": "Legendary",
            },
          ],
          "filterOrder": [
            "Category",
            "Level",
            "Element",
          ],
        },
        "id": 2,
        "itemsAddress": "0x1234567890123456789012345678901234567890",
        "marketplaceType": "market",
        "private": false,
        "projectId": 1,
        "updatedAt": "2025-03-16T13:04:16.098Z",
      },
    ],
    "enabled": true,
    "ogImage": "https://example.com/og-image.png",
    "private": false,
  },
  "projectId": NaN,
  "settings": {
    "accessKey": "mock-access-key",
    "faviconUrl": "https://example.com/favicon.ico",
    "fontUrl": "https://example.com/font.woff2",
    "logoUrl": "https://example.com/logo.png",
    "publisherId": "publisher-1",
    "socials": {
      "discord": "https://discord.gg/mock",
      "instagram": "https://instagram.com/mock",
      "tiktok": "https://tiktok.com/@mock",
      "twitter": "https://twitter.com/mock",
      "website": "https://mock.com",
      "youtube": "https://youtube.com/mock",
    },
    "style": {},
    "title": "Mock Marketplace",
    "walletOptions": {
      "connectors": [],
      "includeEIP6963Wallets": true,
      "oidcIssuers": {},
      "walletType": "UNIVERSAL",
    },
  },
  "shop": {
    "bannerUrl": "https://example.com/shop-banner.png",
    "collections": [
      {
        "bannerUrl": "https://example.com/shop-banner.png",
        "chainId": 1,
        "createdAt": "2025-03-16T13:04:16.098Z",
        "customTokenIds": [],
        "id": 1,
        "itemsAddress": "0x0000000000000000000000000000000000000000",
        "marketplaceType": "shop",
        "name": "Mock Shop Collection",
        "private": false,
        "projectId": 1,
        "saleAddress": "0x0000000000000000000000000000000000000000",
        "saleBannerUrl": "https://example.com/shop-sale-banner.png",
        "tokenIds": [
          "1",
          "2",
          "3",
        ],
        "updatedAt": "2025-03-16T13:04:16.098Z",
      },
      {
        "bannerUrl": "https://example.com/polygon-shop-banner.png",
        "chainId": 137,
        "createdAt": "2025-03-16T13:04:16.098Z",
        "customTokenIds": [],
        "id": 2,
        "itemsAddress": "0x1234567890123456789012345678901234567890",
        "marketplaceType": "shop",
        "name": "Polygon Shop Collection",
        "private": false,
        "projectId": 1,
        "saleAddress": "0x1234567890123456789012345678901234567890",
        "saleBannerUrl": "https://example.com/polygon-shop-sale-banner.png",
        "tokenIds": [
          "4",
          "5",
          "6",
        ],
        "updatedAt": "2025-03-16T13:04:16.098Z",
      },
    ],
    "enabled": true,
    "ogImage": "https://example.com/shop-og-image.png",
    "private": false,
  },
}
`;
